File: CodeForces/1899b/main.py
def inp():
    return(int(input()))
def inlt():
    return(list(map(int,input().split())))
def insr():
    return(input().strip())

def solve() :
    n = inp()
    arr = inlt()

    pref = [0] * (n + 1)

    for i in range(n) :
        pref[i + 1] = pref[i] + arr[i]
    
    ans = 0

    for k in range(1, n + 1) :
        mx = -1
        mn = float("inf")
        if(n % k == 0) :
            for i in range(k, n + 1, k) :
                s = pref[i] - pref[i-k]
                mx = max(mx, s)
                mn = min(mn, s)

        ans = max(ans, mx - mn)
    
    print(ans)
    return
    

t = inp()

for _ in range(t) :
    solve()

File: CodeForces/1899b/main.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

ll pref[3000005];

void solve() {
  ll n, ans = 0;
  cin >> n;

  for (int i = 1, x; i <= n; i++) {
    cin >> x;
    pref[i] = pref[i - 1] + x;
  }

  for (int i = 1; i <= n; i++) {
    if (n % i == 0) {
      ll mx = 0, mn = 2e18;
      for (ll j = i; j <= n; j += i) {
        mx = max(mx, pref[j] - pref[j - i]);
        mn = min(mn, pref[j] - pref[j - i]);
      }
      ans = max(ans, mx - mn);
    }
  }
  cout << ans << "\n";
}

int main() {
  int t;
  cin >> t;
  while (t--) {
    solve();
  }
  return 0;
}